src/webVPython.js

```diff
@@ -222,6 +233,15 @@ class VernierGDX extends EventTarget {
     this._resetGraph();
   }
 
+  /**
+   * a function to set the number of samples to collect based on the period and samplesPerSecond
+   * @returns {null}
+   */
+  _setSamplesToCollect() {
+    if(this._collectFor < 0) return;
+    this._samplesToCollect = Math.floor(this.samplesPerSecond * this._collectFor);
```

Comment:
Probably not critical, but we take a sample at t=0 so, we actually collect (this.samplesPerSecond * this._collectFor) + 1